### DISCUSSION TITLE
Open a file and keep tree open, but another hotkey for open file and close the tree afterwards

### DISCUSSION THREAD

**Discussion by mangelozzi** [nvim-tree/nvim-tree.lua#1980] [Q&A]
I found there is an option: `nvim-tree.actions.open_file.quit_on_open`, but the problem with this it always closes the tree afterwards.
I wish to have `o` perform as default (keep the tree open after opening the file), but have `<CR>` open the buffer and close the tree.

Currently these seems to be the only options (tried them all):
```
`<CR>`            edit                open a file or folder; root will cd to the above directory
`o`
`<2-LeftMouse>`
`<C-e>`           edit_in_place       edit the file in place, effectively replacing the tree explorer
`O`               edit_no_picker      same as (edit) with no window picker
```
`edit_in_place`  is similar but then I have 2 windows after opening the file, and have to go to the other window and close it.

Looking for something like this:
```
require("nvim-tree").setup({
    view = {
        mappings = {
            list = {
                { key = "<CR>", action = "quit_on_open" },
            },
        },
    },    
})
````

**Comment by alex-courtis** [comment]
Moved to #1984
